components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/SQLQueries.java

```diff
@@ -308,7 +308,22 @@
 
     public static final String RETRIEVE_ROLES_OF_SCOPE = "SELECT IOS.ROLES FROM IDN_OAUTH2_SCOPE IOS WHERE SCOPE_KEY" +
                                                         " = ?";
-    public static final String RETRIEVE_PKCE_TABLE = "SELECT PKCE_MANDATORY, PKCE_SUPPORT_PLAIN FROM IDN_OAUTH_CONSUMER_APPS LIMIT 1";
+
+    public static final String RETRIEVE_PKCE_TABLE_MYSQL = "SELECT PKCE_MANDATORY, PKCE_SUPPORT_PLAIN FROM " +
+            "IDN_OAUTH_CONSUMER_APPS LIMIT 1";
+
+    public static final String RETRIEVE_PKCE_TABLE_DB2SQL = "SELECT PKCE_MANDATORY, PKCE_SUPPORT_PLAIN FROM " +
+            "IDN_OAUTH_CONSUMER_APPS FETCH FIRST 1 ROWS ONLY";
+
+    public static final String RETRIEVE_PKCE_TABLE_MSSQL = "SELECT TOP 1 PKCE_MANDATORY, PKCE_SUPPORT_PLAIN FROM " +
+            "IDN_OAUTH_CONSUMER_APPS";
+
+    public static final String RETRIEVE_PKCE_TABLE_INFORMIX = "SELECT FIRST 1 PKCE_MANDATORY, PKCE_SUPPORT_PLAIN FROM " +
+            "IDN_OAUTH_CONSUMER_APPS";
+
+    public static final String RETRIEVE_PKCE_TABLE_ORACLE = "SELECT PKCE_MANDATORY, PKCE_SUPPORT_PLAIN FROM " +
```

Comment:
This looks like wrong. This should wrapped with "SELECT * FROM (....) WHERE ROWNUM < 2